external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py

```diff
@@ -58,38 +58,29 @@ def __init__(self, task_environment: TaskEnvironment):
             self._labels = task_environment.get_labels(include_empty=True)
         else:
             self._labels = task_environment.get_labels(include_empty=False)
+        
+        self._multiclass = False
+        self._multilabel = False
+        self._hierarchical = False
+
         self._multilabel = len(task_environment.label_schema.get_groups(False)) > 1 and \
                            len(task_environment.label_schema.get_groups(False)) == \
                            len(task_environment.get_labels(include_empty=False))  # noqa:E127
 
-        self._hierarchical = False
+        self._hierarchical_info = None
         if not self._multilabel and len(task_environment.label_schema.get_groups(False)) > 1:
             self._hierarchical = True
-            torchreid_env = self.convert_to_torchreid_env(task_environment)
-            self.torchreid_train_task = OTEClassificationTrainingTask(torchreid_env)
-
-    @staticmethod
-    def convert_to_torchreid_env(task_env):
-        model_template = task_env.model_template
-        rename_dict = {'model-preparation-algorithm': 'deep-object-reid',
-                       'classification' : 'ote_custom_classification',
-                       '_cls_incr' : ''
-                       }
-
-        for key, val in rename_dict.items():
-            model_template.model_template_path = model_template.model_template_path.replace(key, val)
-        model_template.entrypoints.base = 'torchreid_tasks.train_task.OTEClassificationTrainingTask'
-        model_template.framework = 'OTEClassification v1.2.3'
-        return task_env
+            self._hierarchical_info = get_hierarchical_info(task_environment.label_schema)
+        
+        if not self._multilabel and not self._hierarchical:
+            self._multiclass = True
```

Comment:
I don't think we need this variable, what do you think? self._multilabel and self._hierarchical is False, the remaining cases are It's true that it's a multi-class, but it doesn't seem very useful.